### ISSUE TITLE
CUDA memory cache increasing

### ISSUE THREAD

**Issue by nepoyasnit** [resemble-ai/resemble-enhance#37]
I've run gradio app from repo and saw that CUDA memory grows rapidly from ~3 Gbs to ~12Gbs. 
If i put small audio, it is also increasing, but not so much. 

CUDA memory when i put small (1 second) audio file:
![image](https://github.com/user-attachments/assets/fd49791a-849f-4fc7-a9db-3ef30bc8b6ce)

CUDA memory after putting long audio file (2.5 minutes):
![image](https://github.com/user-attachments/assets/5b9347b4-5d96-4def-a1b7-d8d20124dc23)

Also, I've checked torch.cuda.memory_allocated() and it was constant, but torch.cuda.memory_cached() was increasing. 
Maybe someone can explain me, why CUDA cache is growing?

**Comment by Cooperos** [comment]
> Also, I've checked torch.cuda.memory_allocated() and it was constant, but torch.cuda.memory_cached() was increasing. Maybe someone can explain me, why CUDA cache is growing?

When you free a tensor, the memory is not returned to the GPU immediately. Instead, it is cached by PyTorch to be reused for future allocations. This is why `torch.cuda.memory_cached()` can increase over time, even if `torch.cuda.memory_allocated()` remains constant.

[Pytorch docs](https://pytorch.org/docs/stable/notes/cuda.html#cuda-memory-management)

**Comment by nepoyasnit** [comment]
> > Also, I've checked torch.cuda.memory_allocated() and it was constant, but torch.cuda.memory_cached() was increasing. Maybe someone can explain me, why CUDA cache is growing?
> 
> When you free a tensor, the memory is not returned to the GPU immediately. Instead, it is cached by PyTorch to be reused for future allocations. This is why `torch.cuda.memory_cached()` can increase over time, even if `torch.cuda.memory_allocated()` remains constant.
> 
> [Pytorch docs](https://pytorch.org/docs/stable/notes/cuda.html#cuda-memory-management)

Yes, but it is strange that with the growth of requests, the size of the enhancer increases so much. Also, i think it's strange to have 13Gb CUDA memory, when only ~3 of that is allocated.

**Comment by aperture147** [comment]
For some reason with some audio input this model allocates 10-13G of memory, which overloads my 10G RTX3080. Is there any way to fix this since calling `torch.cuda.empty_cache()` does not help solving the issue. Thanks!
